nca/NetworkConfig/NetworkLayer.py

```diff
@@ -422,36 +421,30 @@ def _allowed_xgress_conns_optimized(self, is_ingress, peer_container, res_conns_
 class IngressNetworkLayer(NetworkLayer):
 
     def _allowed_xgress_conns(self, from_peer, to_peer, is_ingress):
-        allowed_conns = ConnectionSet()
         all_allowed_conns = ConnectionSet(True)
-        captured_res = False
-        if not is_ingress:
-            allowed_conns, _, _, captured_res = self.collect_policies_conns(from_peer, to_peer, is_ingress)
-            if captured_res:
-                all_allowed_conns = allowed_conns
+        allowed_conns, _, _, captured_res = self.collect_policies_conns(from_peer, to_peer, is_ingress)
+        if captured_res:
+            all_allowed_conns = allowed_conns
         return PolicyConnections(captured=captured_res, allowed_conns=allowed_conns, denied_conns=ConnectionSet(),
                                  all_allowed_conns=all_allowed_conns)
 
     def _allowed_xgress_conns_optimized(self, is_ingress, peer_container, res_conns_filter=PolicyConnectionsFilter()):
-        res_conns = OptimizedPolicyConnections()
-        all_peers_and_ips = peer_container.get_all_peers_group(True)
-        all_peers_no_ips = peer_container.get_all_peers_group()
+        # TODO - should we add non-captured connections from mesh to dns_entries??
+        all_peers_and_ips = peer_container.get_all_peers_group(add_external_ips=True, include_dns_entries=True)
```

Comment:
Why do you include `dns_entries` here but not in non-optimized solution?

Reply:
In non-optimized solution: `all_allowed_conns = ConnectionSet(True)` in case of non-captured, which is equivalent to the above optimized solution.